cfgov/regulations3k/models/pages.py

```diff
@@ -230,23 +230,31 @@ class RegulationPage(RoutablePageMixin, SecondaryNavigationJSMixin, CFGOVPage):
         ObjectList(CFGOVPage.settings_panels, heading='Configuration'),
     ])
 
-    @cached_property
-    def section_query(self):
+    def get_effective_version(self, request, date_str):
+        """ Get the requested effective version if the user has permission """
+        effective_version = self.regulation.versions.get(
```

Comment:
Since we allow users to copy and (probably) import versions that may have the same effective date, `get` could raise `MultipleObjectsReturned`

Reply:
I think I'd rather make effective_date unique on the model. That would eliminate the possibility of multiple versions with the same date.